src/lib.rs

```diff
@@ -1,19 +1,15 @@
-/// # Fantom/libhash
-///
-/// This crate defines a trait which contains commonly used behaviours for hashing.
-/// An example of an implementation of this trait can be found in the libhash-sha3 crate:
-/// (https://github.com/Fantom-foundation/libhash-sha3).
+//! This crate defines a trait which contains commonly used behaviours for hashing.
+//! An example of an implementation of this trait can be found in the [`libhash-sha3`](https://github.com/Fantom-foundation/libhash-sha3) crate.
 use crate::errors::Error;
 use serde::Serialize;
 
-/// The Hashing trait. So far, the trait requires an array of 8 bit unsigned integers as a defined
-/// type.
+//! The Hashing trait. So far, the trait requires an array of 8-bit unsigned integers as a defined
```

Comment:
I tested what happens if we have module comments in the body of the file with `cargo doc` and it errored:
```
error: expected outer doc comment
 --> src/lib.rs:6:1
  |
6 | //! The Hashing trait. So far, the trait requires an array of 8-bit unsigned integers as a defined
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: inner doc comments like this (starting with `//!` or `/*!`) can only appear before items

error: aborting due to previous error

error: Could not document `libhash`.
```
I'll fix this small omission.